.github/workflows/pipeline-build.yml

```diff
@@ -26,11 +27,38 @@ jobs:
           uses: better0fdead/actions-setup-docker@better0fdead/increase-timeout
 
         #Build without test on Windows or macOS (when not a PR)
+        - name: Set up JDK 17
+          uses: actions/setup-java@v4
+          with:
+            distribution: 'temurin'
+            java-version: 17
+
+        - name: Cache SonarQube packages
+          uses: actions/cache@v4
+          with:
+            path: ~/.sonar/cache
+            key: ${{ runner.os }}-sonar
+            restore-keys: ${{ runner.os }}-sonar
+
+        - name: Cache Gradle packages
+          uses: actions/cache@v4
+          with:
+            path: ~/.gradle/caches
+            key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+            restore-keys: ${{ runner.os }}-gradle
+
         - name: Gradle build
+          env:
+            SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+            SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
```

Comment:
@SaeedRe I guess we can remove this, now that it's been hardcoded.

Reply:
Removed the line